pkg/admission/types.go

```diff
@@ -31,3 +32,9 @@ type WantsInformerFactory interface {
 	SetInformerFactory(informers.SharedInformerFactory)
 	Validator
 }
+
+// WantsAuthorizer defines a function which sets Authorizer for admission plugins that need it.
+type WantsAuthorizer interface {
+	SetAuthorizer(authorizer.Authorizer)
+	Validator
```

Comment:
> I don't get how that follows. Can't you just as well type assert to a Validator for that purpose without having to pollute these interface definitions? I would much rather have interface definitions be as narrow as possible.

Seems like that would require the admission plugin author to have remembered to write the `Validate` method.  Since the problem is that we had authors forgetting to do that, I'm not sure having them remember to do a parallel task would help.

Even if a `Validate` method always returns "fine", it's still better to force an admission plugin author to consider the problem.  In some cases the information is optional, in other cases it isn't.